misc/python/materialize/cli/ci_closed_issues_detect.py

```diff
@@ -184,7 +190,9 @@ def main() -> int:
         ):
             issue_refs.extend(detect_closed_issues(filename))
 
-    if args.changed_lines_only:
+    if args.changed_lines_only or (
+        args.changed_lines_only_when_in_pr and not buildkite.is_on_default_branch()
```

Comment:
`buildkite.is_on_default_branch()` because `is_in_pull_request()` has the comment "Note that this does not work in (manually triggered) nightly builds because they don't carry this information!".
The downside is that this will handle release branches like PRs.

Not too happy about this. Maybe we should introduce a `is_likely_in_pull_request()`, which does some heuristics?